### ISSUE TITLE
Upload files using a glob pattern

### ISSUE THREAD

**Issue by danielc2013** [stelligent/crossing-go#8]
Background:
There is another ticket which is supposed to allow for multiple file uploads. It would also be beneficial to allow for regex/wildcards and the ability to preserve a directory structure in S3.

Depends On: https://github.com/stelligent/crossing-go/issues/7

ToDo:
Extend the existing upload functionality to also accept glob-type patterns (*,?,[a-Z], etc) and upload all the files matching that pattern. Multiple patterns should also be supported, and if duplicates are found, only upload once
Example:
`crossing-go put -k alias/crossing-go-key -b s3://my-big-bucket ./*.go ./*.py`
